openassessment/xblock/static/js/src/lms/oa_response.js

```diff
@@ -463,25 +488,46 @@ OpenAssessment.ResponseView.prototype = {
 
     /**
      When selecting a file for upload, do some quick client-side validation
-     to ensure that it is an image, and is not larger than the maximum file
-     size.
+     to ensure that it is an image, a PDF or other allowed types, and is not
+     larger than the maximum file size.
 
      Args:
         files (list): A collection of files used for upload. This function assumes
             there is only one file being uploaded at any time. This file must
-            be less than 5 MB and an image.
+            be less than 5 MB and an image, PDF or other allowed types.
+        uploadType (string): uploaded file type allowed, could be none, image,
+            file or custom.
 
      **/
-    prepareUpload: function(files) {
+    prepareUpload: function(files, uploadType) {
         this.files = null;
-        this.imageType = files[0].type;
+        this.fileType = files[0].type;
+        var ext = files[0].name.split('.').pop().toLowerCase();
+
         if (files[0].size > this.MAX_FILE_SIZE) {
             this.baseView.toggleActionError(
-                'upload', gettext("File size must be 5MB or less.")
+                'upload',
+                gettext("File size must be 5MB or less.")
             );
-        } else if (this.imageType.substring(0,6) !== 'image/') {
+        } else if (uploadType === "image" && this.data.ALLOWED_IMAGE_MIME_TYPES.indexOf(this.fileType) === -1) {
             this.baseView.toggleActionError(
-                'upload', gettext("File must be an image.")
+                'upload',
+                gettext("You can upload files with these file types: ") + "JPG, PNG or GIF"
```

Comment:
I would interpolate these values because string concatenation is a bad idea with translated strings. I'm also wondering why it hard codes the list of valid image file types. I would think these should be configurable somewhere.